src/js/jsx/shared/Datalist.jsx

```diff
@@ -603,7 +626,18 @@ define(function (require, exports, module) {
                 });
             }
 
-            if (this.props.startFocused && this.refs.textInput) {
+            var currentOptions,
+                id;
+
+            if (this.refs.select) {
+                currentOptions = this.refs.select._getOptions();
+            }
+            
+            if (currentOptions && currentOptions.size === 1) {
+                id = currentOptions.toJS()[0].id;
```

Comment:
Calling `.toJS()` is very slow. It would be better to not do so if possible. It seems like you could just do `currentOptions.first().id` instead.